src/tplr/neurons.py

```diff
@@ -167,33 +188,57 @@ async def catchup_with_aggregation_server(
     # Apply aggregation for each step, checking for current window changes
     current_step = checkpoint_window
     while current_step < target_window:
-        # Check if current_window has changed during processing
-        if instance.current_window > target_window:
-            target_window = instance.current_window
-            logger.info(
-                f"Current window advanced during catchup, new target: {target_window}"
-            )
-
         logger.info(
             f"\nProcessing catchup for window {current_step} (Target: {target_window})"
         )
 
-        # Load aggregation for current window - pass version explicitly
+        # Load aggregation for current window
         agg_data = await instance.comms.load_aggregation(window=current_step)
 
+        # For the last window in catchup, we might need to retry a few times
+        if agg_data is None and current_step == target_window - 1:
+            max_retries = 5
+            retry_count = 0
+            retry_delay = 10
+
+            logger.info(
+                f"No aggregation for latest window {current_step}, will retry up to {max_retries} times"
+            )
+
+            while retry_count < max_retries and agg_data is None:
+                retry_count += 1
+                logger.info(
+                    f"Retry {retry_count}/{max_retries} for window {current_step}"
+                )
+                await asyncio.sleep(retry_delay)
+
+                # Try to load aggregation again
+                agg_data = await instance.comms.load_aggregation(window=current_step)
+
+                if agg_data is not None:
+                    logger.info(
+                        f"Successfully loaded aggregation on retry {retry_count}"
+                    )
+                    break
```

Comment:
no need to break here, the loop will exit because `agg_dada is not None` anymore